src/components/Drawer/Drawer.js

```diff
@@ -1,55 +1,112 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
+import { Transition } from 'react-transition-group';
 import bem from 'bem';
 import { IoIosArrowUp } from 'react-icons/io';
 import styles from './Drawer.scss';
 
+const TRANSITION_DURATION = {
+    enter: 10,
+    exit: 300,
+};
+
 const { block, elem } = bem({
     name: 'Drawer',
     classnames: styles,
-    propsToMods: ['isOpen'],
+    propsToMods: ['isShownAndExpanded', 'isShownAndClosed', 'isExpanded'],
 });
 
 const Drawer = props => {
-    const { title, initialIsOpen, children, ...rest } = props;
-    const [isOpen, setIsOpen] = useState(initialIsOpen);
+    const {
+        title,
+        isShow,
```

Comment:
do we really need this? If the component doesn't need to be shown, why to render it at all? Cannot it be part of the business logic, and simply assume that if the component is there, it should be shown indeed (as it is with any other component)? That would simplify the API as well as the state calculation.

Reply:
Yep, I first did this logic in Search, but for the animation we need to keep this in the component. It allows us to make a smooth transition between show/hide states. And I don't fully agree with that is a part of the business logic. It is like a Modal component, it also has the same behavior.

Reply:
I missed the part where there is an animation on the initial render. Thanks